rasa/core/processor.py

```diff
@@ -652,13 +652,9 @@ async def trigger_external_user_uttered(
     @staticmethod
     def _log_slots(tracker: DialogueStateTracker) -> None:
         # Log currently set slots
-        slot_values = "\n".join(
-            [f"\t{s.name}: {s.value}" for s in tracker.slots.values()]
-        )
-        if slot_values.strip():
-            structlogger.debug(
-                "processor.slots.log", slot_values=copy.deepcopy(slot_values)
-            )
+        slots = {s.name: s.value for s in tracker.slots.values() if s.value is not None}
+
+        structlogger.debug("processor.slots.log", slots=slots)
```

Comment:
This is a PII leaking log, so we have to make sure this is compatible with PII:  the dict passed to PII pipeline should take the format of `SlotSet` event as dict (because of this if check expecting `event` key: https://github.com/RasaHQ/rasa-plus/blob/0f8fe716e207b20ba50f144cc25ee3fe27fe2820/rasa_plus/anonymization/anonymization_pipeline.py#L142). Orrr serialize the slots dict - could be the easiest way.

Reply:
mhm I am not sure yet I understand what you mean. the value for the `slots` parameter is a dict right now. are you suggesting to pass a string rather than that dict?

Reply:
mhm I mean this goes against the idea of using structured logging - as the whole point of that is to retain structure rather than flattening to str.

can't we do something like this in pii:

```python
def clean_pii_from_dict(my_dict):
    stripped = strip_pii(json.dumps(my_dict))
    try:
        cleaned_dict = json_loads(stripped)
        return cleaned_dict
    except Exception:
        return "REDACTED"
```

Reply:
We can definitely add another if branch to handle this case and convert the dict to a string, I'll create a ticket and make the squad aware that we'll need extra testing for this before the GA release.